client/my-sites/domains/domain-management/components/domain/new-domains-redirection-notice-upsell.jsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { localize } from 'i18n-calypso';
+import Banner from 'components/banner';
+import { currentUserHasFlag, getCurrentUser } from 'state/current-user/selectors';
+import { NON_PRIMARY_DOMAINS_TO_FREE_USERS } from 'state/current-user/constants';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import isSiteOnPaidPlan from 'state/selectors/is-site-on-paid-plan';
+import { getPlansBySite } from 'state/sites/plans/selectors';
+import { hasCustomDomain } from '../../../../../lib/site/utils';
+
+/**
+ * Style dependencies
+ */
+import './new-domains-redirection-notice-upsell.scss';
+
+class NewDomainsRedirectionNoticeUpsell extends React.Component {
+	render() {
+		const {
+			hasNonPrimaryDomainsFlag,
+			isOnPaidPlan,
+			hasLoadedSitePlans,
+			hasCustomPrimaryDomain,
+			selectedSite,
+			selectedSiteId,
+			translate,
+		} = this.props;
+
+		if (
+			! hasLoadedSitePlans ||
+			! selectedSite ||
+			isOnPaidPlan ||
+			! hasNonPrimaryDomainsFlag ||
+			hasCustomPrimaryDomain
+		) {
+			return null;
+		}
+
+		return (
+			<Banner
```

Comment:
I'd guess this should be the `UpsellNudge` component instead - seee https://github.com/Automattic/wp-calypso/pull/41892 and https://github.com/Automattic/wp-calypso/issues/38778 for related work (cc @sixhours)

Reply:
Nice. Good thing the two components were pretty much interchangeable :)